ee/desktop/runner/runner.go

```diff
@@ -466,6 +467,12 @@ func (r *DesktopUsersProcessesRunner) runConsoleUserDesktop() error {
 		return nil
 	}
 
+	if runtime.GOOS == "linux" && !IsAppindicatorEnabled(context.Background()) {
+		level.Info(r.logger).Log("msg", "no appindicator enabled, not running desktop")
+		r.killDesktopProcesses()
```

Comment:
This feels like we need more research. In my little testing, it seemed like when the screen is locked, the extension is disabled. I'm not sure if we want to _kill_ on that condition, or simply not start

Reply:
it's possibly belt and suspenders, but if someone does disable or install some funky appindicator, I would think we want to turn off the desktop menu to make sure we don't eat CPU

Reply:
What if someone's running cinnamon/mate/xfce4/kde/something else besides gnome?

Reply:
hmm ... I don't know

We could check to see if the user is running gnome and then do the check. Letting non-gnome machines continue to run. 

What do you think?

Reply:
It's been a while since I've tested (and I only tested in multipass VMs, so who knows how accurate of a test), but the desktop seemed to behave okay on non-gnome machines. Maybe worth a quick validation that that's still true? And if so, letting non-gnome machines continue to run desktop.

Reply:
There seems to be lots of ways to check if a machine is using gnome with lots of edge cases, so I installed KDE plasma on a vm and ran `/usr/bin/gnome-extensions list`. It will error with something like "failed to connect to GNOME shell". So I updated the code to just assume that gnome is not running when the cmd fails. I'm not sure if this is good enough

Reply:
I don't know a perfect way to tell if we're running gnome. I think checking for the gnome session, or gdm, probably.

Reply:
I tried a couple of different ways

One machine is just default ubunutu mostly ....
One machine is ubuntu with KDE (plasma) installed after starting up ...

`loginctl`: works in both gnome and kde
`ls /usr/bin/*session`: will list both gnome-session and plasma-session (KDE)
`gdm3 --version`: works on both kde and gnome
`/usr/bin/gnome-extensions list`: fails on KDE works on gnome ... most reliable way I've found so far

Reply:
I was expecting to need to check the process list. 

But, I think just execing `gnome-extensions` might be okay. We're going to exec that anyhow, and if we can discern "not running gnome" than we can call it good enough